File: 0049-group-anagrams/0049-group-anagrams.go
func groupAnagrams(strs []string) [][]string {
    
    mp := map[[26]int][]string{}
    for _ , s := range strs {
        char := [26]int{}
        for _ ,c := range s { 
                char[c-'a']++
            }
            mp[char] = append( mp[char] , s )
        }
        
        ans := [][]string{}
        for _ ,v := range mp { 
            ans = append( ans ,v )
        }
        return ans 
    }

File: 0049-group-anagrams/0049-group-anagrams.cpp
class Solution {
public:
    vector<vector<string>> groupAnagrams(vector<string>& strs) {
        
    unordered_map<string, vector<string>> mp ; 
    for(string s : strs){
        string a = s; 
        sort(a.begin() , a.end()); 
        mp[a].push_back(s) ;
    }
    vector<vector<string>> ans ; 
    for(auto p:mp){
        ans.push_back(p.second) ;
    }
    return ans; 
        
    }
};